lib/ansible/modules/windows/win_file.ps1

```diff
@@ -20,51 +20,601 @@
+                {
+                    linkPath = printName;
+
+                }
+                targets = new string[1];
+                if (linkPath.EndsWith("\\"))
+                {
+                    targets[0] = linkPath.Substring(0, linkPath.Length - 1);
+                }
+                else
+                {
+                    targets[0] = linkPath;
+                }
+
+                type = SYMBOLIC_LINK;
+            }
+            else if (buffer.ReparseTag == IO_REPARSE_TAG_MOUNT_POINT)
+            {
+                string printName = new string(buffer.PathBuffer, (int)(buffer.PrintNameOffset / SIZE_OF_WCHAR), (int)(buffer.PrintNameLength / SIZE_OF_WCHAR));
+                string substituteName = new string(buffer.PathBuffer, (int)(buffer.SubstituteNameOffset / SIZE_OF_WCHAR), (int)(buffer.SubstituteNameLength / SIZE_OF_WCHAR));
+                targets = new string[1];
+                if (printName.EndsWith("\\"))
+                {
+                    targets[0] = printName.Substring(0, printName.Length - 1);
+                }
+                else
+                {
+                    targets[0] = printName;
+                }
+                if (substituteName.StartsWith("\\??\\Volume"))
+                {
+                    type = MOUNT_POINT;
+                }
+                else
+                {
+                    type = JUNCTION_POINT;
+                }
+            }
+            else
+            {
+                string errorMessage = String.Format("Invalid Reparse Tag: {0}, expecting IO_REPARSE_TAG_SYMLINK {1} or IO_REPARSE_TAG_MOUNT_POINT {2}",
+                                        buffer.ReparseTag.ToString(),
+                                        IO_REPARSE_TAG_SYMLINK.ToString(),
+                                        IO_REPARSE_TAG_MOUNT_POINT.ToString());
+                throw new Exception(errorMessage);
+            }
+        }
+
+        private REPARSE_DATA_BUFFER GetReparseData(SafeFileHandle fileHandle)
+        {
+            REPARSE_DATA_BUFFER buffer = new REPARSE_DATA_BUFFER();
+            uint controlCode = FILE_DEVICE_FILE_SYSTEM | FSCTL_GET_REPARSE_POINT;
+            uint bytesReturned;
+            bool success = DeviceIoControl(fileHandle, controlCode, IntPtr.Zero, 0, out buffer, MAXIMUM_REPARSE_DATA_BUFFER_SIZE, out bytesReturned, IntPtr.Zero);
+            if (!success)
+                throw new Exception(String.Format("Failed to get Reparse Point Data Buffer: {0}", new Win32Exception(Marshal.GetLastWin32Error()).Message));
+
+            return buffer;
+        }
     }
 }
 "@
 Add-Type -TypeDefinition $symlink_util
+$symlink_helper = [Ansible.SymLink]
+$symlink_helper::EnableSeBackupPrivilege()
 
 # Used to delete directories and files with logic on handling symbolic links
-function Remove-File($file, $checkmode) {
+function Remove-File($file) {
     try {
         if ($file.Attributes -band [System.IO.FileAttributes]::ReparsePoint) {
```

Comment:
`$file.Attributes.HasFlag([System.IO.FileAttributes]::ReparsePoint)` should also work here.